src/git/remotes/google-source.ts

```diff
@@ -0,0 +1,28 @@
+import { GerritRemote } from './gerrit';
+
+export class GoogleSourceRemote extends GerritRemote {
+	constructor(domain: string, path: string, protocol?: string, name?: string, custom: boolean = false) {
+		super(domain, path, protocol, name, custom, false);
+	}
+
+	override get id() {
+		return 'google-source';
+	}
+
+	override get name() {
+		return this.formatName('Google Source');
+	}
+
+	protected override get baseUrl(): string {
+		return `${this.protocol}://${this.domain}/${this.path}`;
+	}
+
+	private get reviewDomain(): string {
+		const [subdomain, secondLevelDomain, topLevelDomain] = this.domain.split('.');
```

Comment:
FWIW this doesn't work for domains with more than three levels.

e.g. `project.git.source.company-tld.com`

Can this just split and mutate the subdomain?

Reply:
This is only for GoogleSource. Are you sure this is applicable for you?

Reply:
Yes, it was serendipitous that I found this commit because I was just debugging this issue at the time.

Reply:
Upon double-checking, I guess it's not strictly necessary.
The domain I was using also resolves with `project.googlesource.com`

Reply:
> The domain I was using also resolves with project.googlesource.com

Not sure if I understood this part.

Anyway, after this PR, you would configure VS Code with:

```json
"gitlens.remotes": [{ "domain": "project.git.source.company-tld.com", "type": "Gerrit" }]
```

If you have such an environment to test, I would advise you to test it before we merge. The .vsix is available here:

https://github.com/gitkraken/vscode-gitlens/pull/1954#issuecomment-1092309335